Auth module

3ach opened the pull request:
Woo-hoo! A working prototype of the auth module!

3ach committed:
Remove sliver as a dependency

3ach committed:
Add virtual users

3ach committed:
Add new share link backend

3ach committed:
Create alias

3ach committed:
Initial pass at auth

3ach committed:
Add privilege management

3ach committed:
Copy ownedBy into auth table

cjmyers commented:
I cannot open any existing private collections.

> On Oct 4, 2019, at 11:45 PM, Zach Zundel <notifications@github.com> wrote:
> 
> @3ach <https://github.com/3ach> requested your review on: #1019 <https://github.com/SynBioHub/synbiohub/pull/1019> Auth module.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH55HGV6PGA2FOUT5RO3QM7BO7A5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOUBCMW4I#event-2688863089>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA2YH54TYVERUMRLL7LA3ZLQM7BO7ANCNFSM4I5VDFRQ>.
>

3ach commented:
Does that mean collections you already had uploaded don't let you view them? I can't get mine to recreate that. I tried switching back to master, uploading a new collection, then switching to the auth branch and I can open it without issue.

3ach committed:
Update privileges to include sharing

cjmyers commented:
Could it be because I use spoofing on my laptop?  Might make sure you are using right config variable for graphs

Chris

Sent from my iPhone

> On Oct 6, 2019, at 6:34 PM, Zach Zundel <notifications@github.com> wrote:
> 
> Does that mean collections you already had uploaded don't let you view them? I can't get mine to recreate that. I tried switching back to master, uploading a new collection, then switching to the auth branch and I can open it without issue.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or mute the thread.

3ach committed:
Deal with spoofing

3ach commented:
That's probably it. I have uploaded a version which should properly deal with spoofing.

cjmyers commented:
Yep, that fixed it.  I will poke around more and see if I can break anything else :-)

> On Oct 6, 2019, at 7:39 PM, Zach Zundel <notifications@github.com> wrote:
> 
> That's probably it. I have uploaded a version which should properly deal with spoofing.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH56EYL4UDM5YONJJDXTQNIWEZA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOQ5LQ#issuecomment-538775214>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA2YH5ZLFIOMNG7S56YPSG3QNIWEZANCNFSM4I5VDFRQ>.
>

cjmyers commented:
The buttons have expanded with names on them.  That is all except the back to root collection button.  Is this intended?  Seems not really an auth related feature.

> On Oct 6, 2019, at 6:34 PM, Zach Zundel <notifications@github.com> wrote:
> 
> Does that mean collections you already had uploaded don't let you view them? I can't get mine to recreate that. I tried switching back to master, uploading a new collection, then switching to the auth branch and I can open it without issue.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54DGMMCTXD6NBKR5RTQNIOTRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOPPUY#issuecomment-538769363>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA2YH5YZQLOLSOJDLCE27PDQNIOTRANCNFSM4I5VDFRQ>.
>

3ach commented:
Yes I reworked it because I didn’t think a lot of the icons were
particularly clear.

Le lun. 7 oct. 2019 à 15:18, cjmyers <notifications@github.com> a écrit :

> The buttons have expanded with names on them. That is all except the back
> to root collection button. Is this intended? Seems not really an auth
> related feature.
>
> > On Oct 6, 2019, at 6:34 PM, Zach Zundel <notifications@github.com>
> wrote:
> >
> > Does that mean collections you already had uploaded don't let you view
> them? I can't get mine to recreate that. I tried switching back to master,
> uploading a new collection, then switching to the auth branch and I can
> open it without issue.
> >
> > —
> > You are receiving this because your review was requested.
> > Reply to this email directly, view it on GitHub <
> https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54DGMMCTXD6NBKR5RTQNIOTRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOPPUY#issuecomment-538769363>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/AA2YH5YZQLOLSOJDLCE27PDQNIOTRANCNFSM4I5VDFRQ
> >.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AAXTGTKM55LFPJXCEJPO3GTQNORSRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR2YLA#issuecomment-539208748>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAXTGTNIXIJFNPCUSLR4W2LQNORSRANCNFSM4I5VDFRQ>
> .
>

cjmyers commented:
Hmm, well we should probably change them all if we change them.  For that one, you could maybe call it “Return” for return to root collection.

We might also want to separate this though into a separate issue, since it is a separate change.  

> On Oct 7, 2019, at 11:21 PM, Zach Zundel <notifications@github.com> wrote:
> 
> Yes I reworked it because I didn’t think a lot of the icons were
> particularly clear.
> 
> Le lun. 7 oct. 2019 à 15:18, cjmyers <notifications@github.com> a écrit :
> 
> > The buttons have expanded with names on them. That is all except the back
> > to root collection button. Is this intended? Seems not really an auth
> > related feature.
> >
> > > On Oct 6, 2019, at 6:34 PM, Zach Zundel <notifications@github.com>
> > wrote:
> > >
> > > Does that mean collections you already had uploaded don't let you view
> > them? I can't get mine to recreate that. I tried switching back to master,
> > uploading a new collection, then switching to the auth branch and I can
> > open it without issue.
> > >
> > > —
> > > You are receiving this because your review was requested.
> > > Reply to this email directly, view it on GitHub <
> > https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54DGMMCTXD6NBKR5RTQNIOTRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOPPUY#issuecomment-538769363>,
> > or mute the thread <
> > https://github.com/notifications/unsubscribe-auth/AA2YH5YZQLOLSOJDLCE27PDQNIOTRANCNFSM4I5VDFRQ
> > >.
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AAXTGTKM55LFPJXCEJPO3GTQNORSRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR2YLA#issuecomment-539208748>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AAXTGTNIXIJFNPCUSLR4W2LQNORSRANCNFSM4I5VDFRQ>
> > .
> >
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54QPGUU7GNBH5JAWLLQNOZADA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR7XHA#issuecomment-539229084>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA2YH55EIBTZFK4XPCVHVALQNOZADANCNFSM4I5VDFRQ>.
>

3ach commented:
So put in the new sharing button as just an icon and then expand all the
labels in a separate change?

Le lun. 7 oct. 2019 à 16:39, cjmyers <notifications@github.com> a écrit :

> Hmm, well we should probably change them all if we change them. For that
> one, you could maybe call it “Return” for return to root collection.
>
> We might also want to separate this though into a separate issue, since it
> is a separate change.
>
> > On Oct 7, 2019, at 11:21 PM, Zach Zundel <notifications@github.com>
> wrote:
> >
> > Yes I reworked it because I didn’t think a lot of the icons were
> > particularly clear.
> >
> > Le lun. 7 oct. 2019 à 15:18, cjmyers <notifications@github.com> a écrit
> :
> >
> > > The buttons have expanded with names on them. That is all except the
> back
> > > to root collection button. Is this intended? Seems not really an auth
> > > related feature.
> > >
> > > > On Oct 6, 2019, at 6:34 PM, Zach Zundel <notifications@github.com>
> > > wrote:
> > > >
> > > > Does that mean collections you already had uploaded don't let you
> view
> > > them? I can't get mine to recreate that. I tried switching back to
> master,
> > > uploading a new collection, then switching to the auth branch and I can
> > > open it without issue.
> > > >
> > > > —
> > > > You are receiving this because your review was requested.
> > > > Reply to this email directly, view it on GitHub <
> > >
> https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54DGMMCTXD6NBKR5RTQNIOTRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOPPUY#issuecomment-538769363
> >,
> > > or mute the thread <
> > >
> https://github.com/notifications/unsubscribe-auth/AA2YH5YZQLOLSOJDLCE27PDQNIOTRANCNFSM4I5VDFRQ
> > > >.
> > > >
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <
> https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AAXTGTKM55LFPJXCEJPO3GTQNORSRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR2YLA#issuecomment-539208748
> >,
> > > or mute the thread
> > > <
> https://github.com/notifications/unsubscribe-auth/AAXTGTNIXIJFNPCUSLR4W2LQNORSRANCNFSM4I5VDFRQ
> >
> > > .
> > >
> > —
> > You are receiving this because your review was requested.
> > Reply to this email directly, view it on GitHub <
> https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54QPGUU7GNBH5JAWLLQNOZADA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR7XHA#issuecomment-539229084>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/AA2YH55EIBTZFK4XPCVHVALQNOZADANCNFSM4I5VDFRQ
> >.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AAXTGTP5KQTAP6WZDFNDYE3QNO3ALA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEASA6LQ#issuecomment-539234094>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAXTGTIVT4XZ4BBX57HGNEDQNO3ALANCNFSM4I5VDFRQ>
> .
>

cjmyers commented:
Yes, sounds good.  I would like to think about that a bit more.  We do need to come up with a clean design that scales well.  

> On Oct 8, 2019, at 10:51 PM, Zach Zundel <notifications@github.com> wrote:
> 
> So put in the new sharing button as just an icon and then expand all the
> labels in a separate change?
> 
> Le lun. 7 oct. 2019 à 16:39, cjmyers <notifications@github.com> a écrit :
> 
> > Hmm, well we should probably change them all if we change them. For that
> > one, you could maybe call it “Return” for return to root collection.
> >
> > We might also want to separate this though into a separate issue, since it
> > is a separate change.
> >
> > > On Oct 7, 2019, at 11:21 PM, Zach Zundel <notifications@github.com>
> > wrote:
> > >
> > > Yes I reworked it because I didn’t think a lot of the icons were
> > > particularly clear.
> > >
> > > Le lun. 7 oct. 2019 à 15:18, cjmyers <notifications@github.com> a écrit
> > :
> > >
> > > > The buttons have expanded with names on them. That is all except the
> > back
> > > > to root collection button. Is this intended? Seems not really an auth
> > > > related feature.
> > > >
> > > > > On Oct 6, 2019, at 6:34 PM, Zach Zundel <notifications@github.com>
> > > > wrote:
> > > > >
> > > > > Does that mean collections you already had uploaded don't let you
> > view
> > > > them? I can't get mine to recreate that. I tried switching back to
> > master,
> > > > uploading a new collection, then switching to the auth branch and I can
> > > > open it without issue.
> > > > >
> > > > > —
> > > > > You are receiving this because your review was requested.
> > > > > Reply to this email directly, view it on GitHub <
> > > >
> > https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54DGMMCTXD6NBKR5RTQNIOTRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOPPUY#issuecomment-538769363
> > >,
> > > > or mute the thread <
> > > >
> > https://github.com/notifications/unsubscribe-auth/AA2YH5YZQLOLSOJDLCE27PDQNIOTRANCNFSM4I5VDFRQ
> > > > >.
> > > > >
> > > >
> > > > —
> > > > You are receiving this because you were mentioned.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> > https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AAXTGTKM55LFPJXCEJPO3GTQNORSRA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR2YLA#issuecomment-539208748
> > >,
> > > > or mute the thread
> > > > <
> > https://github.com/notifications/unsubscribe-auth/AAXTGTNIXIJFNPCUSLR4W2LQNORSRANCNFSM4I5VDFRQ
> > >
> > > > .
> > > >
> > > —
> > > You are receiving this because your review was requested.
> > > Reply to this email directly, view it on GitHub <
> > https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH54QPGUU7GNBH5JAWLLQNOZADA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAR7XHA#issuecomment-539229084>,
> > or mute the thread <
> > https://github.com/notifications/unsubscribe-auth/AA2YH55EIBTZFK4XPCVHVALQNOZADANCNFSM4I5VDFRQ
> > >.
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AAXTGTP5KQTAP6WZDFNDYE3QNO3ALA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEASA6LQ#issuecomment-539234094>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AAXTGTIVT4XZ4BBX57HGNEDQNO3ALANCNFSM4I5VDFRQ>
> > .
> >
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019?email_source=notifications&email_token=AA2YH5ZXBBGHC5FN4M2XVZDQNT6GJA5CNFSM4I5VDFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAVXOEY#issuecomment-539719443>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA2YH534EYRUYF5TFN4C473QNT6GJANCNFSM4I5VDFRQ>.
>

3ach committed:
Shorten buttons

3ach committed:
Fix setup flow

3ach commented:
Testing --
 - return to page on manage sharing page doesn't actually return
 - needs to show up in shared with me 
 - edit privileges for existing share
 - too many redirects instead of just not showing
 - edits don't actually work (shows login in iframe)
 - spoofing doesn't work on share return to page
 - alias doesn't handle spoofing

3ach committed:
Fix return link on manage sharing page

3ach committed:
Fix back button on share link page

3ach committed:
Fix alias on spoofed instances

cjmyers commented:
Add to Collection for submit is broken with the new auth module.

cjmyers commented:
throw new ERR_HTTP_HEADERS_SENT('set');
    ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at ServerResponse.setHeader (_http_outgoing.js:470:11)
    at ServerResponse.header (/Users/myers/git/synbiohub/node_modules/express/lib/response.js:771:10)
    at ServerResponse.send (/Users/myers/git/synbiohub/node_modules/express/lib/response.js:170:12)
    at ServerResponse.json (/Users/myers/git/synbiohub/node_modules/express/lib/response.js:267:15)
    at ServerResponse.send (/Users/myers/git/synbiohub/node_modules/express/lib/response.js:158:21)
    at Form.form.on (/Users/myers/git/synbiohub/lib/views/submit.js:105:21)
    at Form.emit (events.js:187:15)
    at handleError (/Users/myers/git/synbiohub/node_modules/multiparty/index.js:211:12)
    at IncomingMessage.onReqAborted (/Users/myers/git/synbiohub/node_modules/multiparty/index.js:190:5)
    at IncomingMessage.emit (events.js:182:13)
    at abortIncoming (_http_server.js:444:9)
    at socketOnClose (_http_server.js:437:3)
    at Socket.emit (events.js:187:15)
    at TCP._handle.close (net.js:611:12)

3ach committed:
Show shared on Shared with Me

3ach committed:
Fix sharing page

3ach committed:
Show privilege for share links

3ach committed:
UpdateShare

3ach committed:
Allow editing share privileges

3ach committed:
Show non-collection shared

3ach committed:
Allow edit of shared parts

3ach commented:
I think I have addressed the comments we found in our earlier review of the auth module. I've also rebased it off master to include the OAuth changes and make it mergeable.

3ach committed:
Fix start issue

3ach committed:
Startup bugs

3ach committed:
Fix app.js

3ach committed:
Change addowner to sharing tests

3ach committed:
Fix API tokens and logout

3ach committed:
Fix metadata check

3ach committed:
Fix tests

3ach committed:
Don't die on request abort

cjmyers commented:
Remove menu should be changed back to icon only for now

cjmyers commented:
Exception going to shared with me:

error: (node:44876) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'map' of undefined
    at Object.getShared (/Users/myers/git/synbiohub/lib/auth/access.js:130:23)
    at module.exports (/Users/myers/git/synbiohub/lib/views/shared.js:55:29)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
error: (node:44876) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
error: (node:44876) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

cjmyers commented:
Creating a view only share link shows up in the share settings as view/edit/share

cjmyers commented:
Remove shared ownership then refresh leads to infinite redirections.

cjmyers commented:
Unable to edit public collections with old ownedBy tags

cjmyers commented:
Also cannot edit newly made public submissions

cjmyers commented:
When you make public, edit/share permissions should be migrated to the new URI, view permission can be safely removed.

cjmyers commented:
The ability to make public, you should have share rights and be a curator.

3ach committed:
Fix shared with me

3ach committed:
Fix viewing share link privileges

3ach committed:
Fix infinite redirect

3ach committed:
Update tests

3ach committed:
Remove merge artifact

3ach committed:
Public editing

3ach committed:
Make public access

3ach committed:
Update tests

3ach committed:
Fix make public

3ach commented:
@cjmyers This is ready for your eyes again.

To clear out the auth db, run the following command before starting up SynBioHub:
```
sqlite3 synbiohub.sqlite 'DELETE FROM auth; DELETE FROM SequelizeMeta WHERE name = "006-ownedby-to-auth.js";'
```

cjmyers commented:
Public ownedBy's do not appear to migrate

cjmyers commented:
Shared with me after migration shows some empty items

cjmyers commented:
Warning message on share with user should be mellowed down a bit.

cjmyers commented:
Cannot share public collections, possibly due to problems in migration.

3ach committed:
Fix migration with spoofing

3ach committed:
Fix messaging

3ach committed:
Allow sharing of public items

3ach commented:
I can't get the empty item bug to reproduce.

cjmyers commented:
Ok, will check this on mine.  Is it ready for another test?

3ach commented:
Yes


On Tue, May 26, 2020 at 3:22 PM cjmyers <notifications@github.com> wrote:

> Ok, will check this on mine. Is it ready for another test?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynBioHub/synbiohub/pull/1019#issuecomment-634286550>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXTGTPEO7FNMQGFBYYL54TRTQXJJANCNFSM4I5VDFRQ>
> .
>

cjmyers commented:
I'm running into the same merge issue.  How do I checkout without it wanting to merge, so I can test it?

cjmyers commented:
Here is what I see when I restart below.  I don’t see any public migration.  Also, I still see some blank entires.  Could I perhaps add some more debug somewhere that might help you see why this is happening on my system?

info: Querying https://synbiohub.org/user/myers
info: Querying https://synbiohub.org/user/test
info: Querying https://synbiohub.org/user/igem
info: Querying null
info: Giving myers access to https://synbiohub.org/user/test/test/Exported_seq
info: Giving myers access to https://synbiohub.org/user/test/test/test_collection/1
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation0
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation0/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation1
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation1/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation10
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation10/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation11
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation11/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation12
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation12/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation13
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation13/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation14
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation14/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation15
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation15/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation16
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation16/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation17
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation17/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation18
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation18/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation19
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation19/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation2
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation2/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation20
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation20/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation21
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation21/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation22
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation22/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation23
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation23/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation24
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation24/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation25
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation25/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation26
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation26/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation3
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation3/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation4
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation4/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation5
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation5/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation6
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation6/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation7
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation7/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation8
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation8/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation9
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp/annotation9/range
info: Giving myers access to https://synbiohub.org/user/test/test/pBW842_pCAG_pExp_seq
info: Giving myers access to https://synbiohub.org/user/test/test/test_collection
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: [Java] Received job type org.synbiohub.InitializeJob
info: [Java] Sending result for job ID 1
info: [Java] {"version":"1.8.0_121","jobId":1}
info: Java initialized; JVM 1.8.0_121
info: Giving test access to https://synbiohub.org/user/myers/Nic/Nic_collection/1
info: Giving test access to https://synbiohub.org/user/myers/Nic/Nic_collection
info: Giving test access to https://synbiohub.org/user/myers/Nic/apple
info: Giving test access to https://synbiohub.org/user/myers/Nic/fruit
info: Giving test access to https://synbiohub.org/user/myers/Nic/pear
info: Giving test access to https://synbiohub.org/user/myers/Nic/orange
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation0
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation0/range
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation1
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation1/range
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation2
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation2/range
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation3
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608/annotation3/range
info: Giving test access to https://synbiohub.org/user/myers/testAPI/LN515608_seq
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}
info: {}


> On May 27, 2020, at 9:02 AM, Zach Zundel <notifications@github.com> wrote:
> 
> 
> Delete the old branch with git branch -D auth-module, then pull again
> 
> On Tue, May 26, 2020 at 22:29 cjmyers <notifications@github.com> wrote:
> 
> > I'm running into the same merge issue. How do I checkout without it
> > wanting to merge, so I can test it?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/SynBioHub/synbiohub/pull/1019#issuecomment-634421047>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AAXTGTKGGVUCTSS65XYVO6TRTSJK3ANCNFSM4I5VDFRQ>
> > .
> >
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019#issuecomment-634721726>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA2YH5YDUFZT3X7CK5WW6T3RTUTRDANCNFSM4I5VDFRQ>.
>

cjmyers commented:
Looks like all is good except the confirmation that public is migrating (seems like it might be, since the other issue with sharing seems to work), and the empty objects created by migration.

cjmyers commented:
Doesn't this line in the migration stop public graphs from migrating:

      .filter(ownership => ownership.ownedBy !== graph)

3ach commented:
The `graph` in that case would be `null` since it's the public graph, so it would only remove ownedBys which point to a null user.

cjmyers commented:
I just realized another potential problem.  You are using GetAllOwnedBy.sparql to get all the URIs that are owned by someone, but this could run into the 10k results limit.  Don’t we need to loop this to get all the URIs?

I think this actually explains my non-migration of public.  I have the iGEM dataset with much greater than 10k entries.  This dataset has myself and James as owners.  James is not a user on my system, so you skip migrating these entires.  However, James is the owner of the first 10k entries returned.  Therefore, it never gets to migrating my ownership for the public entries.

cjmyers commented:
I think I also have a hypothesis why the empty links are being created.  I discovered yesterday that our removeCollection does not appear to completely clean out a collection.  There appear to be triples being left behind.  In my instance, there is a deleted Collection that was shared from a test user to me, and I expect that this is causing odd ownerships to be discovered on incomplete records.  

To address this, I think we will need to see if we can refine the query to not pick these up.  We might also want to create a migration scheme to remove these dangling entries.  This would need to be done VERY carefully, of course.  We also need to figure out while removeCollection is leaving remnants.

cjmyers commented:
I just realized another potential problem.  You are using GetAllOwnedBy.sparql to get all the URIs that are owned by someone, but this could run into the 10k results limit.  Don’t we need to loop this to get all the URIs?

> On May 27, 2020, at 11:05 AM, Zach Zundel <notifications@github.com> wrote:
> 
> 
> The graph in that case would be null since it's the public graph, so it would only remove ownedBys which point to a null user.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/SynBioHub/synbiohub/pull/1019#issuecomment-634806561>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA2YH52PCP7L3HOOJLUVZI3RTVB6JANCNFSM4I5VDFRQ>.
>

3ach commented:
That could be the reason for the migration issue.
Remind me, why aren't we automatically looping queries in the SPARQL layer?


On Wed, May 27, 2020 at 11:23 AM cjmyers <notifications@github.com> wrote:

> I think I also have a hypothesis why the empty links are being created. I
> discovered yesterday that our removeCollection does not appear to
> completely clean out a collection. There appear to be triples being left
> behind. In my instance, there is a deleted Collection that was shared from
> a test user to me, and I expect that this is causing odd ownerships to be
> discovered on incomplete records.
>
> To address this, I think we will need to see if we can refine the query to
> not pick these up. We might also want to create a migration scheme to
> remove these dangling entries. This would need to be done VERY carefully,
> of course. We also need to figure out while removeCollection is leaving
> remnants.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SynBioHub/synbiohub/pull/1019#issuecomment-634817024>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXTGTJDAJJ2STI6VSNOHEDRTVEAZANCNFSM4I5VDFRQ>
> .
>

cjmyers commented:
Hmm, it would make sense to loop it in the sparql code when now LIMIT/OFFSET is provided, since otherwise, it ends up with these strange missing results issue.  We do loop for things like fetch/delete/etc., but it may make sense to simply make this part of sparql.js.  This probably should be done, since the maximum number of return entries is something we can change in virtuoso.ini, so we would have variable behavior if people change that otherwise.

cjmyers commented:
Looked into this problem more.  The issue is that the new shared.js seems to assume only Collections can be shared.  It is using getCollectionMetadata to get metadata instead of the query GetTopLevelMetadata.sparql used in the old version.  This means that anything that is not a collection that is shared will not show any metadata.  

The 2nd problem is that during migration, the ownedBy is migrated, but there is also a sbh:canView tag that is not considered.  It appears in my example, the Collection has this tag while the Collection, its members, and their children have the ownedBy tag.  The old shared page would only show the user those with the canView tag.  From there, one could navigate to other pages.  Namely, it shows a starting point of the share tree.  We need something similar here that somehow indicates what is the top of the tree, so it can be displayed on the Shared With Me page, while other objects are accessible but do not need to be displayed on this page.  Any thoughts on how to accomplish this one?

3ach committed:
Add SQL useful for debugging
